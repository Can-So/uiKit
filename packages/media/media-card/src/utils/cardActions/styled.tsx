/* tslint:disable:variable-name */

import styled from 'styled-components';

import { HTMLAttributes, ComponentClass } from 'react';
import { akColorN500 } from '@atlaskit/util-shared-styles';
import { Root, borderRadius, size, center } from '../../styles';

export const Wrapper: ComponentClass<HTMLAttributes<{}>> = styled(Root)`
  display: flex;
  position: relative;
  line-height: 0;
`;

export const CardActionButton: ComponentClass<
  HTMLAttributes<HTMLDivElement>
> = styled.div`
  ${center} ${borderRadius} ${size(26)} color: ${akColorN500};

  &:hover {
    cursor: pointer;
    background-color: rgba(9, 30, 66, 0.06);
  }
`;
