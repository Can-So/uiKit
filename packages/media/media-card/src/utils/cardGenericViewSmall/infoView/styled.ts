/* tslint:disable:variable-name */

import styled from 'styled-components';

import { HTMLAttributes, ComponentClass } from 'react';
import { antialiased, ellipsis } from '../../../styles';
import {
  akGridSizeUnitless,
  akColorN70,
  akColorB300,
  akColorB400,
  akColorB500,
  akColorN900,
} from '@atlaskit/util-shared-styles';

export interface WrapperProps {
  valign?: 'top' | 'bottom';
}

export const Wrapper: ComponentClass<
  HTMLAttributes<{}> & WrapperProps
> = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  ${({ valign }: WrapperProps) => {
    switch (valign) {
      case 'top':
        return 'justify-content: flex-start;';

      case 'bottom':
        return 'justify-content: flex-end;';

      default:
        return 'justify-content: space-around;';
    }
  }};
`;

export const Title: ComponentClass<HTMLAttributes<{}>> = styled.div`
  ${antialiased} ${ellipsis()} font-weight: bold;
  color: ${akColorN900};
  font-size: 12px;
  line-height: 15px;
`;

export const Body: ComponentClass<HTMLAttributes<{}>> = styled.div`
  display: flex;
  align-items: center;
  height: 15px; /* constrain icon height */
  margin-top: 2px;
`;

export const Icon: ComponentClass<HTMLAttributes<{}>> = styled.div`
  display: flex;
  flex-shrink: 0;
  width: ${akGridSizeUnitless * 2.5}px;
  height: ${akGridSizeUnitless * 2}px;
  padding-right: ${akGridSizeUnitless / 2}px;

  & > img {
    width: 100%;
    height: 100%;

    /* prevent default image or alt text from overflowing */
    overflow: hidden;
  }
`;

export interface SubtitleProps {
  isLink?: boolean;
}

export const Subtitle: ComponentClass<
  HTMLAttributes<{}> & SubtitleProps
> = styled.div`
  ${ellipsis()} color: ${akColorN70};
  font-size: 12px;
  line-height: 15px;
  ${({ isLink }: SubtitleProps) => {
    if (!isLink) {
      return '';
    }

    return `
      color: ${akColorB400};

      .media-card:hover & {
        color: ${akColorB300};
        text-decoration: underline;
      }

      .media-card:active & {
        color: ${akColorB500};
      }

    `;
  }};
`;
