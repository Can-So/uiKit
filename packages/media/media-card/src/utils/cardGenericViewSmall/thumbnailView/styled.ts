/* tslint:disable:variable-name */

import styled from 'styled-components';

import { HTMLAttributes, ComponentClass } from 'react';
import { center, centerX, borderRadius, size } from '../../../styles';
import {
  akGridSizeUnitless,
  akColorN20,
  akColorN30,
} from '@atlaskit/util-shared-styles';

const imgSize = 4 * akGridSizeUnitless;

export const RoundedBackground: ComponentClass<HTMLAttributes<{}>> = styled.div`
  ${centerX} ${borderRadius} min-width: ${imgSize}px;
  height: inherit;
  overflow: hidden;
`;

export const LoadingPlaceholder: ComponentClass<
  HTMLAttributes<{}>
> = styled.div`
  ${center} ${size()} color: #cfd4db;
  background-color: ${akColorN20};
`;

export const EmptyPlaceholder: ComponentClass<HTMLAttributes<{}>> = styled.div`
  ${size(imgSize)} ${center} color: #cfd4db;
  background-color: ${akColorN30};
`;
