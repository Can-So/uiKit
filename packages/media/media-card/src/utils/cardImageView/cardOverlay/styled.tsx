/* tslint:disable:variable-name */

import styled from 'styled-components';

import { HTMLAttributes, ComponentClass } from 'react';
import {
  rgba,
  centerX,
  easeOutCubic,
  borderRadius,
  size,
  transition,
  ellipsis,
  absolute,
  antialiased,
} from '../../../styles';
import {
  akColorN70,
  akColorB200,
  akColorN0,
  akColorN800,
  akColorN900,
  akColorB400,
  akColorB300,
} from '@atlaskit/util-shared-styles';

export interface OverlayProps {
  hasError?: boolean;
}

export const TickBox: ComponentClass<
  HTMLAttributes<{}> & OverlayProps
> = styled.div`
  ${size(14)} ${transition()} background-color: ${rgba('#ffffff', 0.5)};
  position: absolute;
  top: 8px;
  right: 8px;
  border-radius: 20px;
  color: #798599; /* TODO FIL-3884: Align color with new design */
  display: flex;
  opacity: 0;

  &.selected {
    opacity: 1;
    color: white;
    background-color: #0052cc; /* TODO FIL-3884: Align with tickbox icons */
  }

  /* Enforce dimensions of "tick" icon */
  svg {
    width: 14px;
  }
`;

export const Overlay: ComponentClass<
  HTMLAttributes<{}> & OverlayProps
> = styled.div`
  ${size()} ${absolute()} ${borderRadius} display: flex;
  justify-content: space-between;
  flex-direction: column;
  background: transparent;
  border: 2px solid transparent;
  transition: 0.3s background ${easeOutCubic}, 0.3s border-color;
  padding: 16px;

  ${({ hasError }: OverlayProps) => {
    if (hasError) {
      return `
        cursor: default;

        &:hover {
          background: transparent;
        }
      `;
    }

    return '';
  }} &:hover, &.active {
    .top-row {
      .title {
        color: ${akColorB400};
      }
    }

    .bottom-row {
      .delete-btn {
        display: flex;
      }
    }
  }

  .file-type-icon {
    display: block;
  }

  &:not(.persistent) {
    &:not(.error):hover {
      background-color: ${rgba(akColorN900, 0.06)};
    }

    &.selectable {
      &.selected {
        background-color: ${akColorB200};

        &:hover {
          /* TODO FIL-3884 add new overlay with rgba(akColorN900, 0.16) */
        }

        .title,
        .bottom-row,
        .file-size,
        .more-btn {
          color: ${akColorN0};
        }
      }
    }
  }

  &.persistent {
    &:not(.active) {
      overflow: hidden;
    }

    .top-row {
      .title {
        transition: opacity 0.3s;
        opacity: 0;
        color: white;
        visibility: hidden;
      }
    }

    .bottom-row {
      opacity: 0;
      transition: transform 0.2s, opacity 0.5s;
      /* This is the height of the overlay footer, needs to be present now since the parent uses flex and 100% doesn't look right anymore */
      transform: translateY(35px);

      .file-type-icon {
        display: none;
      }

      .file-size {
        color: white;
        display: none;
      }

      .more-btn {
        color: ${akColorN0};
        display: none;

        &:hover {
          background-color: rgba(9, 30, 66, 0.2);
        }
      }

      .delete-btn {
        display: none;

        &:hover {
          background-color: rgba(9, 30, 66, 0.2);
        }
      }
    }

    &:hover,
    &.active {
      background-color: ${rgba(akColorN900, 0.5)};

      .title {
        opacity: 1;
        visibility: visible;
      }

      .file-type-icon,
      .file-size {
        display: block;
      }

      .more-btn {
        ${centerX} color: ${akColorN0};
      }

      .delete-btn {
        display: flex;
      }

      .bottom-row {
        opacity: 1;
        transform: translateY(0);
      }
    }

    /* Selectable */
    &.selectable {
      &:hover {
        .tickbox {
          opacity: 1;
        }
      }

      &.selected {
        border-color: ${akColorB200} !important;

        .tickbox {
          background-color: ${akColorB200} !important;
          border-color: ${akColorB200} !important;
          color: white;
        }
      }
    }
  }

  &.error {
    .top-row {
      overflow: visible;
    }
    &:hover,
    &.active {
      .top-row {
        .title {
          color: ${akColorN800};
        }
      }
    }
  }
`;

export const ErrorLine: ComponentClass<HTMLAttributes<{}>> = styled.div`
  height: 24px;
  display: flex;
  align-items: center;
`;

export const LeftColumn: ComponentClass<HTMLAttributes<{}>> = styled.div`
  width: 100%;
  position: relative;
  box-sizing: border-box;
  vertical-align: middle;
`;

export const TopRow: ComponentClass<HTMLAttributes<{}>> = styled.div``;

export const BottomRow: ComponentClass<HTMLAttributes<{}>> = styled.div`
  display: flex;
  align-items: center;
  z-index: 1;
  height: 16px;
`;

export const RightColumn: ComponentClass<HTMLAttributes<{}>> = styled.div``;

export const ErrorMessage: ComponentClass<HTMLAttributes<{}>> = styled.div`
  ${antialiased} display: inline-block;
  vertical-align: middle;
  font-weight: bold;
  color: ${akColorN70};
  font-size: 12px;
  line-height: 15px;
  overflow: hidden;
  max-width: ~'calc(100% - 24px)';
  text-overflow: ellipsis;
  white-space: nowrap;
`;

export const Retry: ComponentClass<HTMLAttributes<{}>> = styled.div`
  ${antialiased} box-sizing: border-box;
  margin-left: 5px;
  font-weight: bold;
  color: ${akColorB400};
  font-size: 12px;

  &:hover {
    cursor: pointer;
    text-decoration: underline;
    color: ${akColorB300};
  }
`;

export const ErrorWrapper: ComponentClass<HTMLAttributes<{}>> = styled.div`
  display: flex;
`;

export const TitleWrapper: ComponentClass<HTMLAttributes<{}>> = styled.div`
  box-sizing: border-box;
  word-wrap: break-word;
  color: ${akColorN800};
  font-size: 12px;
  line-height: 18px;
`;

export const Subtitle: ComponentClass<HTMLAttributes<{}>> = styled.div`
  ${ellipsis('100px')} font-size: 12px;
  color: #5e6c84;
`;

export const Metadata: ComponentClass<HTMLAttributes<{}>> = styled.div`
  display: flex;
  align-items: center;
`;
