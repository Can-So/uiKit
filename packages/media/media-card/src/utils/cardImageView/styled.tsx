/* tslint:disable:variable-name */

import styled from 'styled-components';

import { HTMLAttributes, ComponentClass } from 'react';
import { akColorN20 } from '@atlaskit/util-shared-styles';
import { Root, cardShadow, centerSelf, borderRadius, size } from '../../styles';

export const Wrapper: ComponentClass<HTMLAttributes<{}>> = styled(Root)`
  ${cardShadow} ${borderRadius} background: #fff;
  cursor: pointer;
  line-height: normal;
  position: relative;
  ${size()} .wrapper {
    ${borderRadius} background: ${akColorN20};
    display: block;
    height: inherit;
    position: relative;

    .img-wrapper {
      ${borderRadius} position: relative;
      width: inherit;
      height: inherit;
      display: block;

      img {
        ${centerSelf} max-height: 100%;
        max-width: 100%;
        display: block;
      }
    }
  }
`;
