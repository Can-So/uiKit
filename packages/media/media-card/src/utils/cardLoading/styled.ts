/* tslint:disable:variable-name */
import styled, { keyframes } from 'styled-components';
import { center } from '@atlaskit/media-ui';
import { akColorN20 } from '@atlaskit/util-shared-styles';
import { CardDimensions } from '../..';

export const blinkLoadingAnimation = keyframes`
  0%{
    opacity: 1;
  }

  50%{
    opacity: 0.6;
  }

  100%{
    opacity: 1;
  }
`;

export interface WrapperProps {
  dimensions: CardDimensions;
}

export const Wrapper = styled.div`
  ${center} background: ${akColorN20};
  color: #cfd4db;
  border-radius: inherit;
  max-height: 100%;
  max-width: 100%;
 
  ${(props: WrapperProps) => {
    return `
      width: ${props.dimensions.width};
      height: ${props.dimensions.height};
    `;
  }}
  > span {
    animation: ${blinkLoadingAnimation} 0.8s infinite;
  }
`;
