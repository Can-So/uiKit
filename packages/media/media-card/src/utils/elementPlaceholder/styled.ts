/* tslint:disable:variable-name */

import styled from 'styled-components';

import { HTMLAttributes, ComponentClass } from 'react';
import { akColorN30 } from '@atlaskit/util-shared-styles';
import { borderRadius } from '@atlaskit/media-ui';

export const Wrapper: ComponentClass<HTMLAttributes<{}>> = styled.span`
  ${borderRadius} background-color: ${akColorN30};
`;
