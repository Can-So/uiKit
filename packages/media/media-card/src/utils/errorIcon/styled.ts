/* tslint:disable:variable-name */

import styled from 'styled-components';

import { HTMLAttributes, ComponentClass } from 'react';

export const ErrorIconWrapper: ComponentClass<HTMLAttributes<{}>> = styled.div`
  display: flex;
  color: #ff991f;
`;
