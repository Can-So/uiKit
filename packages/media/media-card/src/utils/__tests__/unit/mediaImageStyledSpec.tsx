import * as React from 'react';
import { shallow } from 'enzyme';
import { ImageViewWrapper } from '../../mediaImage/styled';

describe('MediaImage styled', () => {
  it('default props', () => {
    const element = shallow(<ImageViewWrapper />);
    expect(element).toMatchSnapshot();
  });

  it('isCropped=true', () => {
    const element = shallow(<ImageViewWrapper isCropped={true} />);
    expect(element).toMatchSnapshot();
  });

  it('fadeIn=true', () => {
    const element = shallow(<ImageViewWrapper fadeIn={true} />);
    expect(element).toMatchSnapshot();
  });
});
