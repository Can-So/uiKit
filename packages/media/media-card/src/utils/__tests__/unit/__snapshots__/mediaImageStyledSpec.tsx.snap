// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaImage styled default props 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  object-fit: contain;
}

<img
  className="c0"
/>
`;

exports[`MediaImage styled fadeIn=true 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  object-fit: contain;
  -webkit-animation: bcCCNc .3s cubic-bezier(0.645,0.045,0.355,1);
  animation: bcCCNc .3s cubic-bezier(0.645,0.045,0.355,1);
}

<img
  className="c0"
/>
`;

exports[`MediaImage styled isCropped=true 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

<img
  className="c0"
/>
`;
